realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -724,4 +779,174 @@ public String toString() {
+            if (lastRet < 0) {
+                throw new IllegalStateException("Cannot call remove() twice. Must call next() in between.");
+            }
+            checkConcurrentModification();
+
+            try {
+                RealmList.this.remove(lastRet);
+                if (lastRet < cursor) {
+                    cursor--;
+                }
+                lastRet = -1;
+                expectedModCount = modCount;
+            } catch (IndexOutOfBoundsException e) {
+                throw new ConcurrentModificationException();
+            }
+        }
+
+        final void checkConcurrentModification() {
+            // A Realm ListView is backed by the original Table and not a TableView, this means
+            // that all changes are reflected immediately. It is therefore not possible to use
+            // the same version pinning trick we use for RealmResults (avoiding calling sync_if_needed)
+            // Fortunately a LinkView does not change unless manually altered (unlike RealmResults)
+            // So therefore it should be acceptable to use the same heuristic as a normal AbstractList
+            // when detecting concurrent modifications.
+            if (modCount != expectedModCount) {
+                throw new ConcurrentModificationException();
+            }
+        }
+    }
+
+    private class RealmListItr extends RealmItr implements ListIterator<E> {
+
+        RealmListItr(int index) {
+            if (index >= 0 && index <= size()) {
+                cursor = index;
+            } else {
+                throw new IndexOutOfBoundsException("Starting location must be a valid index: [0, " + (size() - 1) + "]. Index was " + index);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        public boolean hasPrevious() {
+            return cursor != 0;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        public E previous() {
+            checkConcurrentModification();
+            int i = cursor - 1;
+            try {
+                E previous = get(i);
+                lastRet = cursor = i;
+                return previous;
+            } catch (IndexOutOfBoundsException e) {
+                checkConcurrentModification();
+                throw new NoSuchElementException("Cannot access index less than zero. This was " + i + ". Remember to check hasPrevious() before using previous().");
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        public int nextIndex() {
+            return cursor;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        public int previousIndex() {
+            return cursor - 1;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        public void set(E e) {
+            realm.checkIfValid();
+            if (lastRet < 0) {
+                throw new IllegalStateException();
+            }
+            checkConcurrentModification();
```

Comment:
It could, but this order is the same as in ArrayList, so I would prefer keeping it.